### PR TITLE
internal/qmp-gen: initial TestGenerate with StatusInfo type

### PR DESCRIPTION
**Pull request by mdlayher**
Work in progress, but I think this is a good start.

Thoughts?

### CODE REVIEW

**Comment by mdlayher** [comment]
CC @danderson for thoughts as well.  I think this is arguably the easiest way to test this, but would love to hear if you had anything else in mind.

File: internal/qmp-gen/parse_test.go [hunk 1]
```diff
@@ -79,3 +174,33 @@ Got:
...
+	defs, err := readDefinitions(s.URL)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	symbols, err := parse(defs)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	need := neededTypes(symbols)
```
Comment (danderson) [inline internal/qmp-gen/parse_test.go]: If you wanted, you could also not call neededTypes here, which would generate all the symbols in the input text. That'd let you test generating the struct separately from the command. Your call as to whether you want to stick closer to what the codegen does, or if you want to test specific subsets.

**Comment by danderson** [comment]
LGTM also - left a small suggestion, but we could go either way and be totally fine IMO. This looks like a great start.

File: internal/qmp-gen/parse_test.go (hunk 1 above)
Comment (mdlayher) [inline internal/qmp-gen/parse_test.go]: Specific subsets probably makes the most sense for this test.  I'll add an additional test that verifies subsets, and leave this one to verify the "neededTypes" logic.

**Pull request merged**